kyaml/yaml/datamap.go

```diff
@@ -13,11 +13,9 @@ import (
 // SortedMapKeys returns a sorted slice of keys to the given map.
 // Writing this function never gets old.
 func SortedMapKeys(m map[string]string) []string {
-	keys := make([]string, len(m))
-	i := 0
+	keys := make([]string, 0, len(m))
 	for k := range m {
-		keys[i] = k
-		i++
+		keys = append(keys, k)
```

Comment:
I don't think this code change is a functional change, it looks like it does exactly the same thing functionally as it did previously, but I believe your version has slightly worse performance because we are appending to an array rather than having a prebuilt array.

You can test this out in go playground and see that it functionally does the same thing: https://go.dev/play/p/1BBr3XTnHBS

Reply:
I've check the link and the functional did do the same things... the performance should be relativity same as we pre-allocate capacity but I didn't test it so I can't guarantee.

the configmap data is getting sort and loaded here, but I don't know where it get un-sorted when return to the user..